airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/partition_generation_completed_sentinel.py

```diff
@@ -15,3 +15,8 @@ def __init__(self, stream: AbstractStream):
         :param stream: The stream that was processed
         """
         self.stream = stream
+
+    def __eq__(self, other):
```

Comment:
This [test](https://github.com/airbytehq/airbyte/pull/34454/files#diff-d2ec4a1a6427dd0b45b15ed322b8c46c998cb98471ec451b9a2c80264e273bc6L33) was not working as `queue` was a mock and `queue.put.has_calls` was returning a mock. Hence the assertion was passing. In order for the test to work, we need to be able to compare the `PartitionGenerationCompletedSentinel` hence this added method